animation_nodes/nodes/rotation/random_quaternion.py

```diff
@@ -1,28 +1,52 @@
 import bpy
 import random
+import numpy as np
 from bpy.props import *
+from ... utils.math import cantorPair
 from ... events import propertyChanged
 from ... base_types import AnimationNode
+from . c_utils import randomQuaternionList
 
 class RandomQuaternionNode(bpy.types.Node, AnimationNode):
     bl_idname = "an_RandomQuaternionNode"
     bl_label = "Random Quaternion"
 
     nodeSeed: IntProperty(name = "Node Seed", update = propertyChanged, max = 1000, min = 0)
 
+    createList: BoolProperty(name = "Create List", default = False,
+        description = "Create a list of random quaternions",
+        update = AnimationNode.refresh)
+
     def create(self):
-        self.newInput("Integer", "Seed", "seed")
-        self.newOutput("Quaternion", "Quaternion", "randomQuaternion")
+        if self.createList:
+            self.newInput("Integer", "Seed", "seed")
+            self.newInput("Integer", "Count", "count", value = 5, minValue = 0)
+            self.newOutput("Quaternion List", "Quaternions", "randomQuaternions")
+        else:
+            self.newInput("Integer", "Seed", "seed")
+            self.newOutput("Quaternion", "Quaternion", "randomQuaternion")
 
     def draw(self, layout):
-        layout.prop(self, "nodeSeed")
+        row = layout.row(align = True)
+        row.prop(self, "nodeSeed", text = "Node Seed")
+        row.prop(self, "createList", text = "", icon = "LINENUMBERS_ON")
 
     def getExecutionCode(self, required):
-        yield "randomQuaternion = Quaternion(algorithms.random.randomNumberTuple(seed + 24523 * self.nodeSeed, 4, math.pi))"
-        yield "randomQuaternion.normalize()"
+        if self.createList:
+            yield "randomQuaternions = self.randomQuaternions(seed + 24523 * self.nodeSeed, count)"
```

Comment:
Just pass `seed` directly, the whole point of the pairing function is to avoid this expression.